src/main/java/org/mitre/dsmiley/httpproxy/ProxyServlet.java

```diff
@@ -251,7 +251,14 @@ protected void service(HttpServletRequest servletRequest, HttpServletResponse se
     } else {
       proxyRequest = new BasicHttpRequest(method, proxyRequestUri);
     }
-
+    
+    // PATCH
+    if (!preProxyRequest(servletRequest, servletResponse, proxyRequest)) {
```

Comment:
Personally I would name this function
shouldProxyRequest()
to make it clearer since you're asking the subclass if it is going to handle the request itself or if the ProxyServlet should proxy it